cloudconfig/containerinit/container_userdata_test.go

```diff
@@ -84,80 +88,139 @@ func (s *UserDataSuite) SetUpTest(c *gc.C) {
+          nic_name = match.group(1)
+          match = IP_HWADDR.match(ip_line_str)
+          if match is None:
+              continue
+          nic_hwaddr = match.group(1)
+          devices[nic_hwaddr]=nic_name
+      print("found the following devices: " + str(devices))
+      return devices
+
+  def replace_ethernets(interfaces_file, devices):
+      """check if the contents of interfaces_file contain template
+      keys corresponding to hwaddresses and replace them with
+      the proper device name"""
+      interfaces_file_descriptor = open(interfaces_file, "r")
+      interfaces = interfaces_file_descriptor.read()
+      formatter = Formatter()
+      hwaddrs = [v[1] for v in formatter.parse(interfaces) if v[1]]
+      print("found the following hwaddrs: " + str(hwaddrs))
+      device_replacements = dict()
+      for hwaddr in hwaddrs:
+          hwaddr_clean = hwaddr[3:].replace("_", ":")
+          if devices.get(hwaddr_clean, None):
+              device_replacements[hwaddr] = devices[hwaddr_clean]
+      print ("will use the values in:" + str(device_replacements))
+      print("to fix the interfaces file:")
+      print(str(interfaces))
+      formatted = interfaces.format(**device_replacements)
+      print ("into")
+      print(formatted)
+      interfaces_file_descriptor = open(interfaces_file, "w")
+      interfaces_file_descriptor.write(formatted)
+      interfaces_file_descriptor.close()
+
+  ip_output = ip_parse(subprocess.check_output(["ip", "-oneline", "link"]).splitlines())
+  replace_ethernets(INTERFACES_FILE, ip_output)
+  ' > '%[2]s'
+- |2
+
+  if [ -f /usr/bin/python ]; then
+      python /etc/network/interfaces.py
+  else
+      python3 /etc/network/interfaces.py
+  fi
+`[1:]
+
+	s.expectedFallbackConfig = `#cloud-config
 bootcmd:
 - install -D -m 644 /dev/null '%[1]s'
 - |-
   printf '%%s\n' '
-  auto any0 any1 any3 lo
+  auto lo {eth}
 
   iface lo inet loopback
-    dns-nameservers ns1.invalid ns2.invalid
-    dns-search bar foo
-
-  iface any0 inet static
-    address 0.1.2.3/24
-    gateway 0.1.2.1
-
-  iface any1 inet static
-    address 0.2.2.4/24
-    post-up ip route add 0.5.6.0/24 via 0.2.2.1 metric 50
-    pre-down ip route del 0.5.6.0/24 via 0.2.2.1 metric 50
 
-  iface any2 inet dhcp
-
-  iface any3 inet dhcp
-
-  iface any4 inet manual
+  iface {eth} inet dhcp
   ' > '%[1]s'
-runcmd:
-- |-
-  if [ -f %[1]s ]; then
-      echo "stopping all interfaces"
-      ifdown -a
-      sleep 1.5
-      if ifup -a --interfaces=%[1]s; then
-          echo "ifup with %[1]s succeeded, renaming to %[2]s"
-          cp %[2]s %[2]s-orig
-          cp %[1]s %[2]s
-      else
-          echo "ifup with %[1]s failed, leaving old %[2]s alone"
-          ifup -a
-      fi
-  else
-      echo "did not find %[1]s, not reconfiguring networking"
-  fi
-`[1:]
-
-	s.expectedFallbackConfig = `
-auto eth0 lo
+`
+	s.expectedBaseConfig = `
+auto lo {eth}
 
 iface lo inet loopback
 
-iface eth0 inet dhcp
+iface {eth} inet dhcp
```

Comment:
are we leaving raw {eth} in the output file? Or is there something that will turn these into sequential interfaces?